### PR TITLE
build: updates to Micronaut build plugin 5.0.2

### PR DESCRIPTION
**Pull request by sdelamo**
- simplified build
- define group via projectGroup
- test-suite tests are written in java with Junit
- test-suite-kotlin tests are written in kotlin with Junit
- uses internal.bom module

### CODE REVIEW

File: views-bom/build.gradle [hunk 1]
```diff
@@ -1,15 +1,3 @@
-dependencies {
-    constraints {
-        for (Project p : rootProject.subprojects) {
-            if (p.name == "views-bom" || p.name == 'views') continue
-            api "$p.group:micronaut-$p.name:$p.version"
-        }
-
-        runtime "org.freemarker:freemarker:$freemarkerVersion"
```
Comment (sdelamo) [inline views-bom/build.gradle:8]: This bom build was different than the aws bom. Do you know if the right thing is to apply our internal.bom plugin here as well @melix ?

Comment (sdelamo) [reply views-bom/build.gradle:8]: Is the advantage of having these dependencies specified is that if a user uses the views bom the don't need to specify the freemarker dependency version?

```
implementation(platform("io.micronaut.views:micronaut-views-bom:$micronautVersion"))
implementation "org.freemarker:freemarker:$freemarkerVersion"
```

Comment (melix) [reply views-bom/build.gradle:8]: Yes but the correct usage would be:

```
implementation(platform("io.micronaut.views:micronaut-views-bom:$micronautVersion"))
implementation "org.freemarker:freemarker"
```

Note that if those dependencies are migrated to a version catalog, then you'd don't need to declare them in the BOM module (they would use the `managed-` prefix).

Comment (sdelamo) [reply views-bom/build.gradle:8]: I've added the dependencies back to `views-bom/build.gradle` with `api` instead of `runtime`

**Comment by sdelamo** [comment]
@melix can you review this PR ?

**Review by melix — APPROVED** [review]

File: views-bom/build.gradle [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
 }
+
+dependencies {
+    api("org.freemarker:freemarker:$freemarkerVersion")
+    api("com.github.jknack:handlebars:$handlebarsVersion")
+    api("com.fizzed:rocker-runtime:$rockerVersion")
+    api("com.google.template:soy:$soyVersion")
+    api("org.thymeleaf:thymeleaf:$thymeleafVersion")
+    api("org.apache.velocity:velocity-engine-core:$velocityVersion")
+}
```
Comment (rdesgroppes) [inline views-bom/build.gradle:12]: @sdelamo @melix Shouldn't `constraints {` have been preserved here?

I just gave `micronaut-bom:3.2.2` a try (declares `micronaut-views-bom:3.0.2`) and this leads to *much* fatter jars.

Comment (melix) [reply views-bom/build.gradle:12]: Yes, they should be in `constraints`, good catch!

Comment (sdelamo) [reply views-bom/build.gradle:12]: Fixed via https://github.com/micronaut-projects/micronaut-views/pull/331

File: views-bom/build.gradle (hunk 1 above)
Comment (sdelamo) [reply views-bom/build.gradle:8]: Fixed via https://github.com/micronaut-projects/micronaut-views/pull/331

**Pull request merged**